plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -77,12 +81,49 @@ def set_progress(self):
     @property
     def sync_feed(self):
         """
-        :return:    the URL of the feed we should sync
-        :rtype:     str
+        :return:    list the URLs of the feed we should sync
+        :rtype:     list
         """
         repo_url = self.call_config.get(importer_constants.KEY_FEED)
-        if repo_url is not None and not repo_url.endswith('/'):
-            repo_url += '/'
+        if repo_url:
+            repo_url_slash = repo_url
+            self.tmp_dir = tempfile.mkdtemp(dir=self.working_dir)
+            if not repo_url.endswith('/'):
+                    repo_url_slash = repo_url + '/'
+            try:
+                self.check_metadata(repo_url_slash)
+                return [repo_url_slash]
+            except:
+                # treat as mirrorlist
+                return self.parse_as_mirrorlist(repo_url)
+            finally:
+                shutil.rmtree(self.tmp_dir, ignore_errors=True)
+        return [repo_url]
+
+    def parse_as_mirrorlist(self, feed):
+        """
+        :param feed: feed that should be treated as mirrorlist
+        :type:       str
+
+        :return:    list the URLs received from the mirrorlist
+        :rtype:     list
+        """
+        temp_dir = tempfile.mkdtemp(dir=self.working_dir)
+        dst_path = os.path.join(temp_dir, 'urls')
+        downloader = nectar_factory.create_downloader(feed, self.nectar_config)
+        request = DownloadRequest(feed, dst_path)
+        downloader.download_one(request)
+        url_parse = open(dst_path)
+        url_parse = url_parse.read().split('\n')
```

Comment:
And here, you wouldn't need to open the dst_path file. I noticed that it seems like you forgot to close and remove the file, which also wouldn't be necessary with the StringIO approach. You could replace the above with:

```
url_file.seek(0)
url_parse = url_file.read().split('\n')
```

The url_file will automatically get garbage collected when the function returns, so you don't have to worry about closing it or deleting anything.